### ISSUE TITLE
Reproducing the result of Dense Network with FT

### ISSUE THREAD

**Issue by Ldoun** [OPTML-Group/Unlearn-Sparse#4]
First of all, I like your work. It is impressive. 

I'm trying to reproduce your result using Dense Network with the unlearn method FT. However, I wasn't able to reproduce your result using the default Batch size. I could reproduce your result using a batch size of 64 for random data forgetting and a batch size of 512 for Class-wise forgetting. Is this the correct Batch size you used? or did I do something wrong?

Thank You.

**Comment by jinghanjia** [comment]
Thank you for pointing out this. Yes, in our experiments, we utilized the default batch size. I am curious about your findings regarding Class-wise forgetting on Fine-Tuning (FT) with different batch sizes. Could you please share your results? It would be particularly interesting to see how varying the batch size affects the outcomes in this context. 

Thank you~

**Comment by Ldoun** [comment]
Thank you for your response. If the batch size isn't the reason, then the problem for the class-wise forget task might be explained by the high variance in FT-Forget Efficacy.

Could you clarify the default batch size? The arg_parser.py file indicates 256, while Table A1 in C1 of the paper shows 128.
![image](https://github.com/OPTML-Group/Unlearn-Sparse/assets/67096173/eaae2b5e-d0a4-4a69-a809-4286753c798d)

I tested both batch sizes for class-wise forget in 6 independent trials. batch size of 256 reported an average Forget Efficacy Score of 88% and 128 reported an average Forget Efficacy Score of 100%. while a batch size of 512 reported 61%.

(Except for the seed 1, the scores of batch size 256 were too high for my understanding)
![image](https://github.com/OPTML-Group/Unlearn-Sparse/assets/67096173/abdfd0f3-5076-4fb6-9a5b-55ce3eee6310)

And for random data forget with a batch size of 64 reported an average forget efficacy score of 13.24% while retrain reported 13.7%. (Using a Higher batch size resulted in a poor Forget Efficacy Score. batch size of 256 resulted in 3%)

I think this is highly related to a different number of weight updates, which are due to different batch sizes. I also think that similar results can be achieved by scaling the learning rate.

#5 I found that you mentioned learning rate of 1e-4 for GA is only the default for class-wise unlearning. Could you please confirm whether FT uses different learning rates for random data forget and class-wise forget? 

I appreciate your clarification on these points.
Thank you.

**Comment by jinghanjia** [comment]
First and foremost, I want to express my gratitude for the insightful results you've shared. I would like to clarify a point regarding our paper: we have used a default batch size of 256 for our experiments, except for those involving the ImageNet dataset. I apologize for any confusion caused by the typographical errors in Table A1. We will update this part soon.

Regarding the learning rate for FT in the context of random data forgetting, we also use different learning rate compared to class-wise forgetting, which is around 0.04. 

Thank you for your attention to these details again.

**Comment by Ldoun** [comment]
I tested 6 independent trials with lr, the batch size you provided.
Below the gray block is our result
![image](https://github.com/OPTML-Group/Unlearn-Sparse/assets/67096173/ad40eff2-c857-41af-8875-f8de4354a30d)

As you can see I was able reproduced the result with an acceptable margin.
Thank you for your help.
